gino/declarative.py

```diff
@@ -72,6 +73,10 @@ def _init_table(cls, sub_cls):
                     v.name = k
                     columns.append(v)
                     updates[k] = sub_cls.__attr_factory__(v)
+                elif isinstance(v, sa.Index):
+                    inspected_args.append(v)
+                elif sa.Constraint in v.__class__.__mro__:
+                    inspected_args.append(v)
```

Comment:
Do you mean just do ```inspected_args.append(v.copy())```? This works for ```sa.Constraint``` (since it has ha ```copy()``` method). But no such method for ```sa.Index``` unfortunately.

Reply:
Ah, I re-read your comment. I'll add the copy() for the Constraint case. But why is Index different and "not supposed to be copied"?

Reply:
Oh because `sa.Index` does not have `copy()`, and the reason for that I guess is probably that name is a required parameter, and should be unique - copies violate the name uniqueness. So without further handling, current code should suggest "don't define explicit sa.Index in parent class".

Reply:
> Do you mean just do inspected_args.append(v.copy())?

Yes but this is not enough, the copy should also be set in the `updates` dict, so that further reference on the model returns the correct constraint instance.

Reply:
Right! It does seem that for certain constraints (or combinations of constraints) the duplicates are causing some sqlalchemy errors (double commas in emitted SQL, which I guess are genereted from the copied constraints somehow). Maybe the sql alchemy event system is involved here. If I do not copy it seems to work (in the non-inherited case).

I'll finish the tests and update the PR soon so I can get some feedback there (maybe the tests are wrong...).